### PR TITLE
JAMES-2630 propagate reactor objects following blobs-api changes

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java [hunk 1]
```diff
@@ -142,23 +141,23 @@ private PreparedStatement prepareSelect(Session session) {
...
+                .setString(SENDER, mail.getMaybeSender().asString(null))
+                .setList(RECIPIENTS, asStringList(mail.getRecipients()))
+                .setString(ERROR_MESSAGE, mail.getErrorMessage())
+                .setString(REMOTE_ADDR, mail.getRemoteAddr())
+                .setString(REMOTE_HOST, mail.getRemoteHost())
+                .setLong(MESSAGE_SIZE, mail.getMessageSize())
+                .setTimestamp(LAST_UPDATED, mail.getLastUpdated())
+                .setMap(ATTRIBUTES, toRawAttributeMap(mail))
+                .setMap(PER_RECIPIENT_SPECIFIC_HEADERS, toHeaderMap(mail.getPerRecipientSpecificHeaders())))
+            .flatMap(executor::executeVoidReactor);
```
Comment (blackheaven) [inline server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java]: Is there a reason to not directly call `executor.executeVoidReactor`?

Comment (mbaechler) [reply server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java]: yes, `mail.getMessageSize()` can throw, so we want an exception here to be wrapped as a Mono.error.

Comment (blackheaven) [reply server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java]: ok, thanks

Comment (trantienduchn) [reply server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java]: then It's worth to have `CassandraAsynExecutor.executeVoidReactor(Callable<Statement>)` method?

Comment (mbaechler) [reply server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java]: I would say no

**Comment by mbaechler** [comment]
rebased on master

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**